Sources/ApolloCodegenTestSupport/MockFileManager.swift

```diff
@@ -3,77 +3,113 @@ import ApolloCodegenLib
   }
 
-  public var fileExists: fileExistsBlock?
-  public var removeItem: removeItemBlock?
-  public var createFile: createFileBlock?
-  public var createDirectory: createDirectoryBlock?
-  private(set) public var blocksCalled: [BlocksCalled] = []
-
-  /// Initialize the instance with one or many of the method stubs to control the return values.
-  public init(fileExists: fileExistsBlock? = nil,
-              removeItem: removeItemBlock? = nil,
-              createFile: createFileBlock? = nil,
-              createDirectory: createDirectoryBlock? = nil) {
-    self.fileExists = fileExists
-    self.removeItem = removeItem
-    self.createFile = createFile
-    self.createDirectory = createDirectory
+  private func didCall(closure: Closure) {
+    closuresToBeCalled.remove(closure.description)
   }
 
-  public func fileExists(atPath path: String, isDirectory: UnsafeMutablePointer<ObjCBool>?) -> Bool {
-    guard let block = fileExists else {
-      XCTFail("fileExists block must be set before calling this method!")
-      return false
+
+  /// Check whether all mocked closures were called during the lifetime of an instance.
+  public var allClosuresCalled: Bool {
+    return closuresToBeCalled.isEmpty
+  }
+
+  // MARK: FileManager overrides
+
+  public override func fileExists(atPath path: String, isDirectory: UnsafeMutablePointer<ObjCBool>?) -> Bool {
+    let key = #function
+
+    guard
+      let closure = closures[key],
+      case let .fileExists(handler) = closure else {
+        XCTFail("\(key) closure must be set before calling it!")
+        return false
+      }
+
+    defer {
+      didCall(closure: closure)
     }
 
-    blocksCalled.append(.fileExists)
-    return block(path, isDirectory)
+    return handler(path, isDirectory)
   }
 
-  public func removeItem(atPath path: String) throws {
-    guard let block = removeItem else {
-      XCTFail("removeItem block must be set before calling this method!")
-      return
+  public override func removeItem(atPath path: String) throws {
+    let key = #function
+
+    guard
+      let closure = closures[key],
+      case let .removeItem(handler) = closure else {
+        XCTFail("\(key) closure must be set before calling it!")
+        return
+      }
+
+    defer {
+      didCall(closure: closure)
     }
 
-    blocksCalled.append(.removeItem)
-    try block(path)
+    try handler(path)
   }
 
-  public func createFile(atPath path: String, contents data: Data?, attributes attr: FileAttributes?) -> Bool {
-    guard let block = createFile else {
-      XCTFail("createFile block must be set before calling this method!")
-      return false
+  public override func createFile(atPath path: String, contents data: Data?, attributes attr: FileAttributes?) -> Bool {
+    let key = #function
+
+    guard
+      let closure = closures[key],
+      case let .createFile(handler) = closure else {
+        XCTFail("\(key) closure must be set before calling it!")
```

Comment:
You probably want to pass in `#file` and `#line` that this is being called from in the test so it's clear where in a particular test this is failing (ditto for everything else in here)

For what it's worth I've found it to be a good rule of thumb that if you're using an assertion not directly in a `testBlahBlah` method, you should pass in the file and line of the caller so that the error shows up in the test code rather than in the helper code.

Reply:
These are overrides of `FileManager` functions though. Changing the signature they won't match and anything in the `.apollo` namespace extension will no longer route through `MockFileManager`. I'll add another sentence into the error message to indicate the problem lies with `MockFileManager` but I think this is one place we'll have to live without `#file` and `#line`.

Reply:
I've improved the error message to hint at where the problem will be located. Let me know if you're happy with that compromise.

Reply:
ahhhhh right, dammit. compromise LGTM